include/mbedtls/chacha20.h

```diff
@@ -117,15 +117,17 @@ int mbedtls_chacha20_setkey(mbedtls_chacha20_context *ctx,
  *
  * \param ctx       The ChaCha20 context to which the nonce should be bound.
  *                  It must be initialized and bound to a key.
- * \param nonce     The nonce. This must be \c 12 Bytes in size.
+ * \param nonce     The nonce.
+ * \param nonce_length The length of the nonce in Bytes. This may be \c 12 or \c 8 Bytes.
  * \param counter   The initial counter value. This is usually \c 0.
  *
  * \return          \c 0 on success.
  * \return          #MBEDTLS_ERR_CHACHA20_BAD_INPUT_DATA if ctx or nonce is
  *                  NULL.
  */
 int mbedtls_chacha20_starts(mbedtls_chacha20_context *ctx,
```

Comment:
This would be an API breaking change since the Mbed TLS API is still public. If we want to make this change, we need to deprecate the existing API in a minor release and add this updated API in the next major release. I would suggest to instead add a new (potentially internal) API which supports the full range of nonces, and have the existing implementation call it instead. Same comment applies to the ChaChaPoly API update.